pkg/patterns/declarative/reconciler.go

```diff
@@ -138,19 +118,15 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	}
 
 	extraArgs := []string{"--force"}
-	/*
-		if r.options.prune {
-				selectorArg := ""
-				for k, v := range r.labels(name) {
-					if selectorArg != "" {
-						selectorArg += ","
-					}
-					selectorArg += fmt.Sprintf("%s=%s", k, v)
-				}
-
-				extraArgs = append(extraArgs, "--prune", "--selector", selectorArg)
-			}
-	*/
+
+	if r.options.prune {
+		var labels []string
+		for k, v := range r.options.labelMaker(ctx, instance) {
+			labels = append(labels, fmt.Sprintf("%s=%s", k, v))
```

Comment:
Things will go wrong if people choose obnoxious labels (e.g. with a comma in them) but I think that's fine - these labels should be drawn either from the CR instance or hard-coded by the operator.